internal/controller/kyma/controller.go

```diff
@@ -115,62 +114,45 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{RequeueAfter: r.RequeueIntervals.Success}, nil
 	}
 
-	ctx, err := r.getSyncedContext(ctx, kyma)
-
-	if !kyma.DeletionTimestamp.IsZero() && errors.Is(err, remote.ErrAccessSecretNotFound) {
-		logger.Info("access secret not found for kyma, assuming already deleted cluster")
-		r.Metrics.CleanupMetrics(kyma.Name)
-		r.removeAllFinalizers(kyma)
-
-		if err := r.updateKyma(ctx, kyma); err != nil {
-			r.Metrics.RecordRequeueReason(metrics.KymaUnderDeletionAndAccessSecretNotFound, queue.UnexpectedRequeue)
-			return ctrl.Result{}, err
+	if r.SyncKymaEnabled(kyma) {
```

Comment:
can we create an issue to introduce another controller for the separation of single/dual kyma reconciliation?

Reply:
Sure, created it [here](https://github.com/kyma-project/lifecycle-manager/issues/1525). Still needs to be fully refined.